tests/contrib/create_failing_test_for_compare_ids_test.bzl

```diff
@@ -0,0 +1,103 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Implementation of create_failing_test_for_compare_ids_test rule
+def _create_failing_test_for_compare_ids_test_impl(ctx):
+    test_code = """ '
+load("//:compare_ids_test.bzl", "compare_ids_test")
+
+compare_ids_test(
+    name = "test",
+    id = {id},
+    tars = {tars},
+)
+'
+    """.format(
+        id = repr(ctx.attr.id),
```

Comment:
why repr here? ctx.attr.id seems already a string

Reply:
Here repr is effectively adding an extra pair of quotation marks. This is needed since the code that is being formatted needs to have id = "<some_id>" not id = <some_id>.